src/lib.rs

```diff
@@ -113,51 +113,85 @@ where
     type Item = SdrRecord;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let current_id = self.next_id?;
+        let mut recoverable_errors = 0;
+        const MAX_RECOVERABLE_ERRORS: usize = 10; // Max number of recoverable errors
 
-        if current_id.is_last() {
-            self.next_id.take();
-            return None;
-        }
-
-        let next_record = self
-            .ipmi
-            .send_recv(sdr::GetDeviceSdr::new(None, current_id));
-
-        let (value, next_record_id) = match next_record {
-            Ok(record) => {
-                let next_record_id = record.next_entry;
-
-                (Some(record.record), next_record_id)
-            }
-            Err(IpmiError::ParsingFailed {
-                error: ParseResponseError::Parse((e, next_record_id)),
-                ..
-            }) => {
-                log::warn!(
-                    "Recoverable error while parsing SDR record 0x{:04X}: {e:?}",
-                    current_id.value()
-                );
-                (None, next_record_id)
-            }
-            Err(e) => {
-                log::error!(
-                    "Unrecoverable error while parsing SDR record 0x{:04X}: {e:?}",
-                    current_id.value()
-                );
+        while let Some(current_id) = self.next_id {
```

Comment:
Please use `self.next_id.take()` here and only refer to `current_id` to ensure we cannot forget to clear `self.next_id`. That way we will only continue iteration if `self.next_id` is explicitly assigned.